File: src/main/java/no/nav/veilarbaktivitet/aktivitetskort/bestilling/ArenaAktivitetskortBestilling.java
package no.nav.veilarbaktivitet.aktivitetskort.bestilling;

import lombok.Getter;
import no.nav.veilarbaktivitet.aktivitetskort.ActionType;
import no.nav.veilarbaktivitet.aktivitetskort.Aktivitetskort;
import no.nav.veilarbaktivitet.aktivitetskort.AktivitetskortType;
import no.nav.veilarbaktivitet.arena.model.ArenaId;

import java.util.UUID;

@Getter
public class ArenaAktivitetskortBestilling extends AktivitetskortBestilling {
    private final ArenaId eksternReferanseId;
    private final String arenaTiltakskode;

    public ArenaAktivitetskortBestilling(Aktivitetskort aktivitetskort, String source, AktivitetskortType type, ArenaId eksternReferanseId, String arenaTiltakskode, UUID messageId, ActionType actionType) {
        super(source, type, aktivitetskort, messageId, actionType, aktorId, oppfolgingsPeriode);
        this.eksternReferanseId = eksternReferanseId;
        this.arenaTiltakskode = arenaTiltakskode;
    }
}

File: src/main/kotlin/no/nav/veilarbaktivitet/aktivitetskort/bestilling/ArenaAktivitetskortBestilling.kt
package no.nav.veilarbaktivitet.aktivitetskort.bestilling

import no.nav.veilarbaktivitet.aktivitetskort.ActionType
import no.nav.veilarbaktivitet.aktivitetskort.dto.Aktivitetskort
import no.nav.veilarbaktivitet.aktivitetskort.dto.AktivitetskortType
import no.nav.veilarbaktivitet.arena.model.ArenaId
import no.nav.veilarbaktivitet.person.Person
import java.time.ZonedDateTime
import java.util.*

class ArenaAktivitetskortBestilling(
    aktivitetskort: Aktivitetskort,
    source: String,
    type: AktivitetskortType,
    val eksternReferanseId: ArenaId,
    val arenaTiltakskode: String,
    val oppfolgingsperiode: UUID,
    val oppfolgingsperiodeSlutt: ZonedDateTime?,
    messageId: UUID,
    actionType: ActionType,
    aktorId: Person.AktorId,
) : AktivitetskortBestilling(source, type, aktivitetskort, messageId, actionType, aktorId) {

    override fun getAktivitetskortId() = aktivitetskort.id
}
